pyproject.toml

```diff
@@ -111,6 +111,11 @@ omit = [
 
 [tool.coverage.report]
 ignore_errors = true
+exclude_lines = [
+  '\#\s*pragma: no cover',  # we need this because this field overrides the default
+  '^\s*raise NotImplementedError\b',
```

Comment:
Why exclude `raise NotImplementedError`? Error conditions are supposed to be tested as well.

Reply:
Yes but `NotImplementedError` is for something that is supposed to work eventually, but doesn't currently, so it's different from something where we do actually intent on throwing an error. Writting tests to see if something raises `NotImplementedError` doesn't seem very productive, I'm not really sure what actual value it brings on most situations.